### ISSUE TITLE
Restore multiple (different) variables from the same checkpoint tensor.

### ISSUE THREAD

**Issue by smr97** [tensorflow/tensorflow#40404]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.15
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Currently, if I use the `init_from_checkpoint()` with an `assignment_map`, I can't map one checkpoint tensor to multiple separate graph variables. Using a list only works for multiple partitions of the same variable. For my research use-case, I want to restore `n` matrices of the same shape, from a single checkpoint(ed) matrix of the same shape as well. Once initialisation is done however, these `n` matrices have no co-relation with one another. They will be trained and updated completely independent of each other. Hence I don't want the same variable in the graph.

**Will this change the current api? How?**
It may change the semantics of assignment map, I am not sure.

**Who will benefit with this feature?**
Anyone who wants to re-use large pretrained deep learning models, but play around with architectural changes, say for pruning or model compression.

**Any Other info.**
I am wondering if as a workaround, I can have multiple calls to `init_from_checkpoint()`, one for each `i in n`? And restore them one by one basically. Will the `i'th` call "overwrite" the `i-1'th` call?

**Comment by karmel** [comment]
TensorFlow v2 has upgraded many of the saving features, and allows for more robust model saving and reloading. See https://www.tensorflow.org/guide/saved_model for more information. You can also take a look at the pruning features available from the Model Optimization Toolkit: https://www.tensorflow.org/model_optimization

If you have feature requests for the TF v2 APIs, please post a new issue, and we can discuss there.
